lib/rubocop/cop/graphql/extract_type.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module GraphQL
+      class ExtractType < Cop
+        # This cop checks fields on common prefix groups
+        #
+        # # @example
+        #   # good
+        #
+        #   class Types::UserType < Types::BaseObject
+        #     field :registered_at, String, null: false
+        #     field :contact, Types::ContactType, null: false
+        #
+        #     def contact
+        #      self
+        #     end
+        #   end
+        #
+        #   class Types::ContactType < Types::BaseObject
+        #     field :phone, String, null: false
+        #     field :first_name, String, null: false
+        #     field :last_name, String, null: false
+        #   end
+        #
+        #   # bad
+        #
+        #   class Types::UserType < Types::BaseObject
+        #     field :registered_at, String, null: false
+        #     field :contact_phone, String, null: false
+        #     field :contact_first_name, String, null: false
+        #     field :contact_last_name, String, null: false
+        #   end
+        #
+        include RuboCop::GraphQL::NodePattern
+
+        def on_class(node)
+          schema_member = RuboCop::GraphQL::SchemaMember.new(node)
+
+          if (body = schema_member.body)
+            check_fields_prefixes(body)
+          end
+        end
+
+        private
+
+        MSG = "Consider moving %<field_names>s to a new type and adding the `%<prefix>s` field instead"
+
+        def check_fields_prefixes(body)
+          fractured(body).each do |prefix, fields|
+            if fields.count >= cop_config["MaxFields"]
+              add_offense(
+                fields[-1].node,
+                message: message(prefix, fields.map(&:name).join(", "))
+              )
+            end
+          end
+        end
+
+        def fractured(body)
+          body.each_with_object({}) do |node, acc|
+            next acc unless field_definition?(node) || field_definition_with_body?(node)
+
+            field = RuboCop::GraphQL::Field.new(node)
+            next acc unless field.name.to_s.include?("_")
+
+            prefix = field.name.to_s.split("_").first
```

Comment:
Oh, the user can disable cop https://github.com/DmitryTsepelev/rubocop-graphql/issues/8 and then ExtractType cop will be useless. I think we should handle such cases. Do you agree?

Other question: Will it be assumed that words have the same prefix even if they are written in different notations?

For example:
```
field :contact_phone, String, null: false
field :contactFirstName, String, null: false
```
It is necessary to add offense in this case ?

I think so yes, because regardless of the notation, field names semantically have the same prefix.

Reply:
Yeah, that's definitely possible, would be nice to find out what notation is used and take the prefix from it